tsl/src/continuous_aggs/common.c

```diff
@@ -1060,9 +1060,14 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 
 			if (!both_buckets_are_equal)
 			{
-				char *offset = DatumGetCString(DirectFunctionCall1(interval_out, offset_datum));
+				char *offset =
```

Comment:
While this probably works, I am wondering if we shouldn't check this  further up around line 1044.

It seems one of the `bucket_time_offsets` is NULL, so I think we should have NULL check on the original C pointer before trying to convert to a Datum. We should probably not call ` IntervalPGetDatum()` on a NULL pointer to begin with and check the `bucket_time_offset_pointer` instead. One alternative is to also use a `NullableDatum` to pass this info down.

Reply:
Thanks for the review, @erimatnor!

I think using `NullableDatum` would make sense for datums stored by value. Interval is stored as a pointer so we can just compare it with `NULL` directly. And `IntervalPGetDatum` just converts the pointer type to datum, doesn't dereference it. 
```c
#define IntervalPGetDatum(X) PointerGetDatum(X)
```
So while I could implement what you are suggesting, I'm not sure it is really needed here.